netbox_agent/network.py

```diff
@@ -0,0 +1,118 @@
+import os
+import re
+
+from netaddr import IPAddress
+import netifaces
+
+from netbox_agent.config import netbox_instance as nb
+from netbox_agent.ethtool import Ethtool
+
+IFACE_TYPE_100ME_FIXED = 800
+IFACE_TYPE_1GE_FIXED = 1000
+IFACE_TYPE_1GE_GBIC = 1050
+IFACE_TYPE_1GE_SFP = 1100
+IFACE_TYPE_2GE_FIXED = 1120
+IFACE_TYPE_5GE_FIXED = 1130
+IFACE_TYPE_10GE_FIXED = 1150
+IFACE_TYPE_10GE_CX4 = 1170
+IFACE_TYPE_10GE_SFP_PLUS = 1200
+IFACE_TYPE_10GE_XFP = 1300
+IFACE_TYPE_10GE_XENPAK = 1310
+IFACE_TYPE_10GE_X2 = 1320
+IFACE_TYPE_25GE_SFP28 = 1350
+IFACE_TYPE_40GE_QSFP_PLUS = 1400
+IFACE_TYPE_50GE_QSFP28 = 1420
+IFACE_TYPE_100GE_CFP = 1500
+IFACE_TYPE_100GE_CFP2 = 1510
+IFACE_TYPE_100GE_CFP4 = 1520
+IFACE_TYPE_100GE_CPAK = 1550
+IFACE_TYPE_100GE_QSFP28 = 1600
+IFACE_TYPE_200GE_CFP2 = 1650
+IFACE_TYPE_200GE_QSFP56 = 1700
+IFACE_TYPE_400GE_QSFP_DD = 1750
+IFACE_TYPE_OTHER = 32767
+
+# Regex to match base interface name
+# Doesn't match vlan interfaces and other loopback etc
+INTERFACE_REGEX = re.compile('^(eth[0-9]+|ens[0-9]+|enp[0-9]+s[0-9]f[0-9])$')
+
+
+class Network():
+    def __init__(self, server, *args, **kwargs):
+        self.nics = []
+
+        self.server = server
+        self.scan()
+
+    def scan(self):
+        for interface in os.listdir('/sys/class/net/'):
```

Comment:
Unless I'm mistaken, `/sys` is linux specific, is that ok ?

Reply:
We should add BSD support later